web/app/js/components/Namespace.jsx

```diff
@@ -4,12 +4,36 @@ import ErrorBanner from './ErrorBanner.jsx';
 import MetricsTable from './MetricsTable.jsx';
 import PageHeader from './PageHeader.jsx';
 import { processMultiResourceRollup } from './util/MetricUtils.js';
+import PropTypes from 'prop-types';
 import React from 'react';
 import { withContext } from './util/AppContext.jsx';
 import './../../css/list.css';
 import 'whatwg-fetch';
 
 class Namespaces extends React.Component {
+  static defaultProps = {
+    match: {
+      params: {
+        namespace: 'default',
+      },
+    },
+  }
+
+  static propTypes = {
+    api: PropTypes.shape({
```

Comment:
I thought about it a couple times and didn't go through with it. I'm still not sure what the right way (tm) to go about it is. We can generate a propType from the object signature, but then components that only use subsets will have explicit propTypes suggesting they use them. I tried to go the route of "this component uses these and only these" .... but that didn't work out as well as I'd liked either (as you're mentioning).

fwiw. I think this will be a non-issue when all the components are converted to using `withRest`. At that point, they'll no long have api as a propType.

Reply:
Cool, yep, that works for me.